VTS/Core/Implementations/WebSocketNetCore/WebSocketNetCoreImpl.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace VTS.Core
+{
+    public class WebSocketNetCoreImpl : IWebSocket
+    {
+        private static readonly UTF8Encoding Encoder = new UTF8Encoding();
+
+        private ClientWebSocket? _socket;
+        private readonly ConcurrentQueue<string> _intakeQueue;
+        private readonly ConcurrentQueue<Action> _responseQueue;
+        private bool _attemptReconnect;
+
+        private Action _onConnect = () => { };
+        private Action _onDisconnect = () => { };
+        private Action<Exception> _onError = (e) => { };
+
+        private string _url = "";
+        private readonly IVTSLogger _logger;
+
+        public WebSocketNetCoreImpl(IVTSLogger logger)
+        {
+            _logger = logger;
+            _intakeQueue = new ConcurrentQueue<string>();
+            _responseQueue = new ConcurrentQueue<Action>();
+        }
+
+        public string GetNextResponse()
+        {
+            _intakeQueue.TryDequeue(out var response);
+            return response!;
+        }
+
+        public bool IsConnecting()
+        {
+            return _socket is { State: WebSocketState.Connecting };
+        }
+
+        public bool IsConnectionOpen()
+        {
+            return _socket is { State: WebSocketState.Open };
+        }
+
+        public void Send(string message)
+        {
+            var buffer = Encoder.GetBytes(message);
+            _socket?.SendAsync(new ReadOnlyMemory<byte>(buffer), WebSocketMessageType.Text, true, default)
```

Comment:
Type `ReadOnlyMemory` does not exist in .NET Standard 2.0, is it okay to replace with `ArraySegment`?

Reply:
Yep, that should do the trick. I replaced it and it works like a charm.